reasoner_util/__init__.py

```diff
@@ -185,3 +185,45 @@ def merge_qnodes(qnodes1: dict, qnodes2: dict) -> dict:
         elif qnode_value != new_nodes[qnode_key]:
             raise ValueError("Key exists in both messages but values do not match.")
     return new_nodes
+
+
+def merge_attributes(
+        knode1_attributes: List[dict],
+        knode2_attributes: List[dict],
+        in_place: bool = False
+        ) -> List[dict]:
```

Comment:
We use a specific scheme for indenting things here, and a different one below. We should try to be consistent.